lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -270,10 +270,9 @@ module.exports = {
     const oldSubscriptionFilters = await Promise.all(
       response.subscriptionFilters.map(async (subscriptionFilter) => {
         const { destinationArn, filterName } = subscriptionFilter;
-        const functionName = this.getFunctionNameFromDestinationArn(destinationArn);
         const logicalId = this.getLogicalIdFromFilterName(filterName);
         const stackName = await this.provider
-          .request('CloudFormation', 'describeStackResources', { PhysicalResourceId: functionName })
+          .request('CloudFormation', 'describeStackResources', { PhysicalResourceId: filterName })
           .then(({ StackResources }) => StackResources[0].StackName)
           .catch(() => null);
```

Comment:
oh.. or we can use subscription filter's logical id to get stack resource (currently, `describeStackResources` returns all stack resource with the stack resource specified by physical resource id)
I will apply that if possible

Reply:
Please let me know whenever you feel it's ready for a re-review. Thanks for all you work :bow:

Reply:
sorry for distracted comment with late apply :)